tests/src/Kernel/DatabaseLogTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\rdf_entity\Kernel;
+
+use Drupal\Core\Database\Database;
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\Tests\rdf_entity\Traits\RdfDatabaseConnectionTrait;
+
+/**
+ * Tests the query logging.
+ *
+ * @group rdf_entity
+ *
+ * @coversDefaultClass \Drupal\rdf_entity\Database\Driver\sparql\Connection
+ */
+class DatabaseLogTest extends KernelTestBase {
+
+  use RdfDatabaseConnectionTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+    $this->setUpSparql();
+  }
+
+  /**
+   * Tests the log.
+   *
+   * @param string $method
+   *   The method name 'query' or 'update'.
+   * @param string $query
+   *   The query.
+   * @param array $args
+   *   The query arguments.
+   *
+   * @dataProvider provider
+   */
+  public function testLog(string $method, string $query, array $args): void {
+    Database::startLog('log_test', 'sparql_default');
+    $this->sparql->{$method}($query);
+    $log = $this->sparql->getLogger()->get('log_test');
+
+    $this->assertCount(1, $log);
+
+    $log_entry = reset($log);
+    $this->assertEquals($query, $log_entry['query']);
+    $this->assertSame($args, $log_entry['args']);
+    $this->assertEquals('default', $log_entry['target']);
+    $this->assertEquals('double', gettype($log_entry['time']));
+    $this->assertGreaterThan(0, $log_entry['time']);
+    // @todo Inspect also $log_entry['caller'] when
+    // https://www.drupal.org/project/drupal/issues/2867788 lands.
+    // @see https://www.drupal.org/project/drupal/issues/2867788
+  }
+
+  /**
+   * Data provider for ::testLog().
+   *
+   * @return array
+   *   Test cases.
+   *
+   * @see DatabaseLogTest::testLog()
+   */
+  public function provider(): array {
+    return [
+      'query' => [
+        'query',
+        'SELECT DISTINCT ?s ?p ?o WHERE { ?s ?p ?o } LIMIT 100',
+        [],
```

Comment:
Both of the test cases provide an empty array for the `$args`. Can we add a third case that contains some arguments so we can check that this also works with non-empty arguments?

Reply:
This is related to my comment above, the `$args` are in fact never passed to the database connection handler so they are always empty. There is probably little point in testing this.

Reply:
I know, but given this TODO https://github.com/ec-europa/rdf_entity/pull/52/files#diff-a66e7deed432e098655cb2b1f4322667R95, I anticipated that in the future we'll add a the ability to pass arguments to the query (of course with proper sanitization, which now is embed in the PDO). So, I added $args more as a placeholder, acting as a reminder.

Reply:
I added the 3rd test case with arguments.